### DISCUSSION TITLE
Welcome to video.js Discussions!

### DISCUSSION THREAD

**Discussion by gkatsev** [videojs/video.js#7165] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by kinetachien** [comment]
I am unable to get the control icons to show up properly. No matter whether I use the https://vjs.zencdn.net/7.11.4/video-js.css link or link  the css locally plus hosting the fonts. Is this an issue? Is there a trick to this? It makes the player unusable.


![videojsgarbage](https://user-images.githubusercontent.com/38273797/117725175-9f8f1880-b199-11eb-9d9b-b1a186bb19e3.png)

**Reply by gkatsev** [reply]
When a player looks like that it generally means that either the CSS isn't included on the page or that the player div doesn't have the `video-js` class applied to it.

**Reply by kinetachien** [reply]
Thanks! The issue was the class missing. That was easy. Much appreciated.

**Reply by gkatsev** [reply]
See the example with the `<video>` element here https://github.com/videojs/video.js#quick-start. Adding it to your embed is likely the easiest.
So, `<video class="video-js">`
There's some info here too https://docs.videojs.com/tutorial-embeds.html

**Comment by bcolflesh** [comment]
Anybody have a workaround for this? When you use AWS Elastic Transcoder and add a caption file, you'll get a line like this in the m3u8:

X-TIMESTAMP-MAP=MPEGTS:0, LOCAL:00:00:00.000

Note the space in front of LOCAL - video.js can't deal with this and does not display the captions. Editing the file to remove the space, ex:

X-TIMESTAMP-MAP=MPEGTS:0,LOCAL:00:00:00.000

Makes video.js play nice again. I don't hold out hope for AWS doing the right thing - can video.js be altered to ignore/remove the space in this case?

**Reply by gkatsev** [reply]
There is a pending PR out for this, which we hope to get in soon. In the meantime, please open a ticket with AWS anyway.

**Reply by bcolflesh** [reply]
Thanks for the reply, fingers crossed for a video,js solution! I posted on AWS Re:post here:

https://repost.aws/questions/QUMTLnH6pmTVC2G_rAWcEl9A/elastic-transcoder-error-in-x-timestamp-map-output

**Reply by misteroneill** [reply]
The PR in question is here: https://github.com/videojs/vtt.js/pull/60

We'll work on getting that released soon!

**Reply by bcolflesh** [reply]
Thanks so much, this is super important for us!

**Reply by mrrizwan304** [reply]
Yes, you can address this issue by customizing video.js to handle the space in the X-TIMESTAMP-MAP header. You can create a plugin or modify the library to preprocess the m3u8 file and remove any spaces in the X-TIMESTAMP-MAP line before processing.

Here's a high-level approach:

Intercept the Manifest: Use video.js's Tech or HLS tech events to intercept the manifest file.
Modify the Manifest: Implement a function to parse the manifest and remove spaces from X-TIMESTAMP-MAP lines.
Update video.js: Ensure the modified manifest is fed back into video.js for playback.
If modifying video.js is not feasible, you might consider pre-processing the m3u8 files on your server to remove spaces from X-TIMESTAMP-MAP lines before serving them.

**Comment by hunterboy77766** [comment]
Can we edit any of our videos and post them on this web, can this happen with our[ Alight Motion App ](https://thealightmotionapk.com/)?

**Comment by PedjaCucak** [comment]
Hi everyone,

Is it possible to turn off automatic apply of Closed Caption tracks (never mind that they have DEFAULT and AUTOSELECT)?

If in manifest I have something like this:

#EXT-X-MEDIA:TYPE=CLOSED-CAPTIONS,GROUP-ID="Closed Captions",NAME="en",DEFAULT=YES,AUTOSELECT=YES,LANGUAGE="en",INSTREAM-ID="CC1"

Video.js will automatic show this track on screen, is there a way to turn this off, I don't want player to show track automatic, never mind it has default and autoselect set to true?

Is there some property with which I can configure player not to show CC tracks automatically?
Show them only when user select CC track manually.

Thanks in advance,
Pedja

**Reply by PedjaCucak** [reply]
@gkatsev

**Comment by ChathuraGH** [comment]
Is there any community websites or forums where related people discuss about videojs and related stuff?

**Reply by amtins** [reply]
The slack community link https://slack.videojs.com/

**Reply by ChathuraGH** [reply]
@amtins Thanks.

**Comment by ChathuraGH** [comment]
I have some videojs plugins on my local environment developed by me. I have no idea how to publish them. Any advice? Thanks.

**Comment by shadee-ux** [comment]
Anyone help me with videojs player file with dash drm activated

**Comment by ChathuraGH** [comment]
```
bundles test/**/*.test.js → test/dist/bundle.js...
[!] (plugin babel) BrowserslistError: [BABEL] E:\aa\Completed\bb\v2\test\plugin.test.js: Unknown version 64 of op_mob (While processing: "E:\\aa\\Completed\\bb\\v2\\node_modules\\@babel\\preset-env\\lib\\index.js")
test/plugin.test.js

```


**Dose anybody have any ideas how to fix this issue ?**

**Reply by ChathuraGH** [reply]
Ok. I found what was happening. I was running `npm audit fix` that updated some packages breaking the build process. Don't run `npm audit fix`.
